Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -361,3 +390,26 @@ getOverduePaymentCache endTime = Hedis.get (mkOverduePaymentProcessingKey endTim
 
 setOverduePaymentCache :: CacheFlow m r => UTCTime -> NominalDiffTime -> m ()
 setOverduePaymentCache endTime expTime = Hedis.setExp (mkOverduePaymentProcessingKey endTime) False (round expTime)
+
+buildLocation :: (MonadFlow m) => Id DLocation.Location -> LatLong -> DLocation.LocationAddress -> m DLocation.Location
```

Comment:
this code duplicated in startRide and endRide.  Can be moved to `Domain.Types.Location `